docs/narr/advfeatures.rst

```diff
@@ -0,0 +1,399 @@
+bunch of routes, you can just use the ``include`` statement with a
+``route_prefix``. All the routes of that application will be availabe, prefixed
+as you requested:
+
+.. code-block:: python
+   :linenos:
+
+   from pyramid.config import Configurator
+
+   if __name__ == '__main__':
+      config = Configurator()
+      config.include('pyramid_jinja2')
+      config.include('pyramid_exclog')
+      config.include('some.other.package', route_prefix='/somethingelse')
+
+.. seealso::
+
+    See also :ref:`including_configuration` and
+    :ref:`building_an_extensible_app`.
+
+Authenticate Users Your Way
+---------------------------
+
+:app:`Pyramid` ships with prebuilt well-tested authentication and authorization
+schemes out of the box. Using a scheme is a matter of configuration. So if you
+need to change approaches later, you need only update your configuration.
+
+In addition, the system that handles authentication an authorization is
+flexible and pluggable. If you want to use another security add-on, or define
+your own, you can. And again, you need only update your application
+configuration to make the change.
+
+Example: :ref:`enabling_authorization_policy`.
+
+Build Trees of Resources
+------------------------
+
+:app:`Pyramid` supports :term:`Traversal`, a way of mapping URLs to a concrete
+tree of resources. If your application naturally consists of an arbitrary
+heirarchy of different types of content (like a CMS or a Document Management
+System), traversal is for you. If you have a requirement for a highly granular
+security model ("Jane can edit documents in *this* folder, but not *that*
+one"), traversal can be a powerful approach.
+
+Examples: :ref:`hello_traversal_chapter` and
+:ref:`much_ado_about_traversal_chapter`.
+
+Take Action on Each Request with Tweens
+---------------------------------------
+
+Pyramid has a system for applying arbitrary actions to each request or response
+called *tweens*. The system is similar in concept to WSGI :term:`middleware`,
+but can be more useful since they run in the Pyramid context, and have access
+to templates, request objects, and other niceties.
+
+The Pyramid debug toolbar is a "tween", as is the ``pyramid_tm`` transaction
+manager.
+
+Example: :ref:`registering_tweens`.
+
+Return What You Want From Your Views
+------------------------------------
+
+We have shown before (in the :doc:`introduction`) how using a :term:`renderer`
+allows you to return simple Python dictionaries from your view code. But some
+frameworks allow you to return strings or tuples from view callables.
+When frameworks allow for this, code looks slightly prettier, because there are
+fewer imports, and less code. For example, compare this:
+
+.. code-block:: python
+   :linenos:
+
+   from pyramid.response import Response
+
+   def aview(request):
+       return Response("Hello world!")
+
+To this:
+
+.. code-block:: python
+   :linenos:
+
+   def aview(request):
+       return "Hello world!"
+
```

Comment:
Remember, the first indent is actually not a code indent, it's there to make this code part of the `.. code-block::` directive above.  You are right that there are three spaces there.  It's intended to be so.  In the rendered document that first line is up against the left margin.